forte/data/ontology/ontology_code_generator.py

```diff
@@ -1078,6 +1079,38 @@ def parse_entry(
 
         return entry_item, property_names
 
+    def parse_ndarray(
+        self,
+        manager: ImportManager,
+        schema: Dict,
+        entry_name: EntryName,
+        att_name: str,
+        att_type: str,
+        desc: str,
+    ):
+        if SchemaKeywords.ndarray_value_type not in schema:
+            raise TypeNotDeclaredException(
+                f"Item type for the attribute {att_name} of the entry "
+                f"[{entry_name.class_name}] not declared. This attribute is "
+                f"a composite type: {att_type}, it should have a "
+                f"{SchemaKeywords.ndarray_value_type}."
```

Comment:
This exception is not used properly. `TypeNotDeclaredException` is used when the ontology spec uses a certain type, but we don't know what type it is (hence not declared).

For example, the ontology `Sentence` uses the type `SpecialToken`, but we don't know what `SpecialToken` is.